src/ar/mangaswat/src/eu/kanade/tachiyomi/extension/ar/mangaswat/MangaSwat.kt

```diff
@@ -95,105 +105,228 @@ class MangaSwat :
+            chapters.addAll(data.results.map { chapterFromDto(it) })
+            nextPage = data.next
         }
 
+        return chapters
+    }
+
+    private fun chapterFromDto(chapter: ChapterDto): SChapter = SChapter.create().apply {
+        name = chapter.chapter
+        date_upload = runCatching { apiDateFormat.parse(chapter.createdAt)?.time }.getOrNull() ?: 0L
+        url = "/chapters/${chapter.id}/${chapter.slug}/"
+    }
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        val id = chapter.url.removePrefix("/chapters/").substringBefore("/")
+        val apiHeaders = headersBuilder()
+            .add("Accept", "application/json, text/plain, */*")
+            .add("Origin", baseUrl)
+            .add("Referer", "$baseUrl/")
+            .build()
+        return GET("$apiBaseUrl/chapters/$id/", apiHeaders)
+    }
+
+    override fun pageListParse(response: Response): List<Page> {
+        val chapter = json.decodeFromString<PageListResponse>(response.body.string())
+        return chapter.images.map {
+            Page(it.order, imageUrl = it.image)
+        }
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
         if (query.isNotBlank()) {
-            urlBuilder
-                .removePathSegment(0)
-                .removeAllQueryParameters("title")
-                .addQueryParameter("s", query)
+            val apiHeaders = headersBuilder()
+                .add("Accept", "application/json, text/plain, */*")
+                .add("Origin", baseUrl)
+                .add("Referer", "$baseUrl/")
                 .build()
+            return GET("$apiBaseUrl/series/?search=$query&page=$page", apiHeaders)
         }
 
-        return request.newBuilder()
-            .url(urlBuilder.build())
-            .build()
+        return super.searchMangaRequest(page, query, filters)
     }
 
-    override val orderByFilterOptions = arrayOf(
-        Pair(intl["order_by_filter_default"], ""),
-        Pair(intl["order_by_filter_az"], "a-z"),
-        Pair(intl["order_by_filter_za"], "z-a"),
-        Pair(intl["order_by_filter_latest_update"], "update"),
-        Pair(intl["order_by_filter_latest_added"], "added"),
-        Pair(intl["order_by_filter_popular"], "popular"),
+    override fun searchMangaParse(response: Response): MangasPage {
+        if ("appswat.com" !in response.request.url.host) {
+            return super.searchMangaParse(response)
+        }
+
+        val data = json.decodeFromString<LatestUpdatesResponse>(response.body.string())
+        val mangas = data.results.map {
+            SManga.create().apply {
+                title = it.title
+                thumbnail_url = it.poster.mediumUrl
+                url = "/series/id/${it.id}/"
+            }
+        }
+        return MangasPage(mangas, data.next != null)
+    }
+
+    @Serializable
+    private data class MangaDetailsDto(
+        @SerialName("title") val title: String,
```

Comment:
The serialized name is the same as the variable name, so you can remove it. Also, make it private by moving the `SManga` or `SChapter` creation to a DTO function

```suggestion
        private val title: String,
```